src/main/java/org/seqdoop/hadoop_bam/IndexedBAMInputFormat.java

```diff
@@ -0,0 +1,116 @@
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.lib.input.FileSplit;
+import org.seqdoop.hadoop_bam.util.WrapSeekable;
+
+public class IndexedBAMInputFormat
+    extends BAMInputFormat {
+
+    private static Path filePath;
+    private static Path indexFilePath;
+    private static String viewContig;
+    private static int viewStart;
+    private static int viewEnd;
+    private static SAMSequenceDictionary dict;
+
+    public static void setVars(Path bamPath,
+                               Path baiPath,
+                               String contig,
+                               int start,
+                               int end,
+                               SAMSequenceDictionary sd) {
+        filePath = bamPath;
+        indexFilePath = baiPath;
+        viewContig = contig;
+        viewStart = start;
+        viewEnd = end;
+        dict = sd;
+    }
+
+    @Override public RecordReader<LongWritable, SAMRecordWritable>
+        createRecordReader(InputSplit split,
+                           TaskAttemptContext ctx) 
+        throws IOException, InterruptedException {
+        RecordReader<LongWritable, SAMRecordWritable> rr = new BAMFilteredRecordReader();
+        BAMFilteredRecordReader.setRegion(viewContig, viewStart, viewEnd);
+        rr.initialize(split, ctx);
+        return rr;
+    }
+
+    @Override public List<InputSplit> getSplits(JobContext job)
+        throws IOException, FileNotFoundException {
+
+        Configuration conf = ContextUtil.getConfiguration(job);
+        FileSystem fs = indexFilePath.getFileSystem(conf);
+
+        if (!fs.exists(indexFilePath)) {
+            throw new java.io.FileNotFoundException("Bam index file not provided");
+        } else {
+            WrapSeekable seekableIdxFile = WrapSeekable.openPath(fs,
+                                                                 indexFilePath);
+
+            // Use index to get the chunks for a specific region, then use them to create InputSplits
+            PublicDiskBasedBAMFileIndex idx = new PublicDiskBasedBAMFileIndex(seekableIdxFile,
+                                                                              dict);
+            int referenceIndex = dict.getSequenceIndex(viewContig);
+
+            // Get the chunks in the region we want (chunks give start and end file pointers into a BAM file)
+            SAMFileSpan sfs = idx.getSpanOverlapping(referenceIndex, viewStart, viewEnd);
+            List<Chunk> regions = PublicBAMFileSpan.getChunks(sfs);
+            Iterator<Chunk> ri = regions.iterator();
+            List<InputSplit> splits = new LinkedList<InputSplit>();
```

Comment:
Another case to think about (possibly as a follow on) is what if the intervals are very large - much larger than an HDFS block (e.g. a whole chromosome). Then we'd want the to break up the split roughly on HDFS block boundaries, just like  BAMInputFormat does already.